common/str.h

```diff
@@ -224,6 +235,23 @@ class String {
 	uint hash() const;
 
 	/**
+	 * Various replace functions to suit different parameters.
+	 * 
+	 * For now, they are unable to change the size of the string.
+	 * Thus, they do not insert, they replace the characters.
+	 */
+	// Replace 'count' bits, starting from 'pos' with str[pos, pos+count)
+	void replace(uint32 pos, uint32 count, const String& str);
+	// The same as above, but accepts a C-like array of characters	
+	void replace(uint32 pos, uint32 count, const char* str);
+	// Replace the space between the iterators with str[begin, end)
```

Comment:
IMO, this is inaccurate. It should be something along the lines of:

// Replace the characters in [begin, end) with str[0, end - begin)